extensions/pkg/controller/dnsrecord/reconciler.go

```diff
@@ -250,18 +251,15 @@ func updateCreatedCondition(status extensionsv1alpha1.Status, conditionStatus ga
 	if c != nil && c.Status == conditionStatus {
 		return nil
 	}
-	if c == nil {
-		conditions = append(conditions, gardencorev1beta1helper.InitCondition(extensionsv1alpha1.ConditionTypeCreated))
-		c = &conditions[len(conditions)-1]
-	}
 
 	builder, err := gardencorev1beta1helper.NewConditionBuilder(extensionsv1alpha1.ConditionTypeCreated)
-	if err != nil {
-		return err
+	utilruntime.Must(err)
```

Comment:
[Don't panic](https://github.com/golang/go/wiki/CodeReviewComments#dont-panic) in controllers for handling errors.

Reply:
done